include/mq/api/Abilities.h

```diff
@@ -0,0 +1,200 @@
+ * innate skills, but innate skills must be specified as their "stacked"
+ * (above 100) number.
+ *
+ * @param nSkill The skill identifier
+ *
+ * @return bool True if the character has the specified skill, false otherwise
+ **/
+MQLIB_API bool HasSkillOrInnate(int nSkill);
+
+/**
+ * @fn SkillOrInnateIsActivatable
+ *
+ * @brief Checks if the skill is activatable
+ *
+ * Returns whether the skill is activatable. An activatable skill
+ * is also known as an Ability. This includes innate skills, but
+ * innate skills must be specified as their "stacked" (above 100)
+ * number.
+ *
+ * @param nSkill The skill identifier
+ *
+ * @return bool True if the skill is activatable, false otherwise
+ **/
+MQLIB_API bool SkillOrInnateIsActivatable(int nSkill);
+
+/**
+ * @fn GetSkillName
+ *
+ * @brief Retrieves the name of a skill based on its identifier
+ *
+ * Returns the name of a skill given its identifier. This includes
+ * innate skills, but innate skills must be specified as their
+ * "stacked" (above 100) number.
+ *
+ * @param nSkill The skill identifier
+ *
+ * @return const char* The name of the skill, or nullptr if an issue
+ **/
+MQLIB_API const char* GetSkillName(int nSkill);
+
+/**
+ * @fn GetSkillIDFromString
+ *
+ * @brief Converts a string to a corresponding skill identifier
+ *
+ * Attempts to convert a string into an skill identifier. If the string
+ * represents a numeric value, it is converted directly to an integer.
+ * If it's a non-numeric string, the function searches for a matching
+ * skill name and returns its identifier.
+ *
+ * For innate abilities, the function will only return the "stacked"
+ * (above 100) value.
+ *
+ * The function returns a default value (iReturnOnFail) if the string is
+ * empty, if no matching skill is found, or if the numeric conversion
+ * results in a value outside the valid range of skill identifiers.
+ *
+ * @param svString The string representing the skill, either as a name or a number
+ * @param iReturnOnFail (Optional) The value to return on failure to convert, defaults to -1
+ *
+ * @return int The skill identifier corresponding to the string, or iReturnOnFail on failure
+ **/
+MQLIB_API int GetSkillIDFromString(std::string_view svString, int iReturnOnFail = -1);
+
+/**
+ * @fn GetAbilityIDFromString
+ *
+ * @brief Converts a string to a corresponding ability identifier
+ *
+ * Attempts to convert a string into an ability identifier using GetSkillIDFromString.
+ * An ability is a skill that can be activated, so if the skill is not activatable,
+ * this will return a failure.
+ *
+ * @see GetSkillIDFromString
+ *
+ * @param svString The string representing the ability, either as a name or a number
+ * @param iReturnOnFail (Optional) The value to return on failure to convert, defaults to -1
+ *
+ * @return int The ability identifier corresponding to the string, or iReturnOnFail on failure
+ **/
+MQLIB_API int GetAbilityIDFromString(std::string_view svString, int iReturnOnFail = -1);
```

Comment:
GetAbilityIDFromString or GetAbilityIDByString? hmm..

Reply:
From has parallels and this can include an int lookup as well as the name.  ByString would make more sense to me if it was name only...but then it would be GetAbilityIDByName.